### PR TITLE
COMP: add completion for `clippy::all`

### PR DESCRIPTION
**Pull request by Kobzol**
This PR adds the completion for the "virtual" `clippy::all` group. I also updated the lints while we're at it.

Fixes: https://github.com/intellij-rust/intellij-rust/issues/8931

changelog: The `clippy::all` lint group will now be completed.

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/completion/lint/RsClippyLintCompletionProvider.kt [hunk 1]
```diff
@@ -13,7 +13,7 @@ import org.rust.lang.core.psi.RsPath
 
 object RsClippyLintCompletionProvider : RsLintCompletionProvider() {
     override val prefix: String = "clippy::"
-    override val lints: List<Lint> = CLIPPY_LINTS
+    override val lints: List<Lint> = CLIPPY_LINTS.plus(Lint("all", true))
```
[the remark this answers is not in the document]
Comment (Kobzol) [reply src/main/kotlin/org/rust/lang/core/completion/lint/RsClippyLintCompletionProvider.kt:16]: I was thinking about it, but then the list in the build script would have to be converted into a `MutableList`, and I would need to construct a JSON value containing the proper parameters manually.. it seemed to be much more complicated than this change. But if you want I'll do it in the script.

[the remark this answers is not in the document]
Comment (Undin) [reply src/main/kotlin/org/rust/lang/core/completion/lint/RsClippyLintCompletionProvider.kt:16]: It seems you are talking about some code in `build.gradle.kts`, but I meant just to add the corresponding value [here](https://github.com/intellij-rust/intellij-rust/blob/9e45ce3181769da9c3539d769e5581d4f2fac2dd/scripts/fetch_lints.py#L69-L70) (and probably check that this item is not already in the list)

[the remark this answers is not in the document]
Comment (Kobzol) [reply src/main/kotlin/org/rust/lang/core/completion/lint/RsClippyLintCompletionProvider.kt]: Oh, good point, that's probably the simplest solution. I moved it to the Python script.

**Review by Undin — APPROVED** [review]
Thanks!

bors r+

**Pull request merged**